src/commands.rs

```diff
@@ -1060,31 +1045,29 @@ pub fn pws_clear(ctx: &mut Context<'_>, slot: u8) -> anyhow::Result<()> {
 
 fn print_pws_slot(
   ctx: &mut Context<'_>,
-  pws: &nitrokey::PasswordSafe<'_, '_>,
-  slot: usize,
-  programmed: bool,
+  index: usize,
+  slot: Option<nitrokey::PasswordSlot<'_, '_, '_>>,
 ) -> anyhow::Result<()> {
-  let slot = u8::try_from(slot).map_err(|_| anyhow::anyhow!("Invalid PWS slot number"))?;
-  let name = if programmed {
-    pws
-      .get_slot_name(slot)
-      .context("Failed to read PWS slot name")?
+  let name = if let Some(slot) = slot {
+    slot.get_name().context("Failed to read PWS slot name")?
   } else {
     "[not programmed]".to_string()
   };
-  println!(ctx, "{}\t{}", slot, name)?;
+  println!(ctx, "{}\t{}", index, name)?;
   Ok(())
 }
 
 /// Print the status of all PWS slots.
 pub fn pws_status(ctx: &mut Context<'_>, all: bool) -> anyhow::Result<()> {
   with_password_safe(ctx, |ctx, pws| {
-    let slots = pws
-      .get_slot_status()
-      .context("Failed to read PWS slot status")?;
+    let slots = pws.get_slots().context("Failed to read PWS slot status")?;
```

Comment:
Consider adjusting the error message slightly to `"Failed to read PWS slots"`.

Reply:
I’m not sure about that.  Internally, we still call the `GET_PW_SAFE_SLOT_STATUS` command.  “Failed to read PWS slots” sounds as if we tried to read the slot values.